### ISSUE TITLE
Verbosity

### ISSUE THREAD

**Issue by AndrasKovacs** [haskell/primitive#188]
I would like to directly use `primitive` arrays in a significant amount of code, and the verbosity of function names is very annoying. 

I don't see a good reason why `newSmallArray` should be preferred to `new`. No other commonly used data structure library has this verbose naming convention. Instead, the standard practice is to let library users disambiguate by qualified imports. The verbose naming scheme is perhaps justified at the level of `GHC.Prim` but not here really.

**Comment by treeowl** [comment]
I agree. I think we should really add a package to this repository using the other convention, as I said elsewhen.

**Comment by cartazio** [comment]
As David says: a wrapper package for that needs be made.  But all the
preexisting code does the no clobber approach

**Comment by chessai** [comment]
The API provided by the `contiguous` package is quite nice: http://hackage.haskell.org/package/contiguous-0.3.3.0/docs/Data-Primitive-Contiguous.html

The benefits of `contiguous`: no needing to remember naming conventions for primitive, no inconsistent API between modules, flexibility in the array type you use

Version 0.4 will be even more uniform and complete (w.r.t. functions), and have support for the fourth array type defined in primitive
